### DISCUSSION TITLE
Installing new Models

### DISCUSSION THREAD

**Discussion by grendahl06** [vladmandic/sdnext#2619] [Q&A] [answered]
TL;DR; through the interface I've downloaded new models from HuggingFace (one for anime and for realism); How do I use them or install them now that I've downloaded them?

To be clear, I used the Models tab, selected HuggingFace, searched for the models, then his download.  I watched in the console as it downloaded several 5GB files.  I tried restarting the server to see if that would help.

Thank you in advance.

By the way, the default settings work perfectly for a Radeon 6650XT.  (Crashes with out of memory when you turn on too many new things...I'm sure that's my fault.)

**Comment by vladmandic** [comment]
a) make sure you're set backend=diffusers as huggingface models only work in diffusers mode (and restart after changing mode)
b) in main interface click button networks -> diffusers -> select model

**Reply by grendahl06** [reply]
in step A, where do I make this change?  

also, as far as you know, if I switch to using SDXL models, should this continue to work for Radeon cards?

thank you

**Reply by grendahl06** [reply]
at the moment, I get the following error:

18:06:31-944173 ERROR    Error loading model weights:
                         D:\sdxl\automatic\models\Stable-diffusion\realistic_mk2.safetensors
18:06:31-946172 ERROR    Error(s) in loading state_dict for LatentDiffusion:     size mismatch for
                         model.diffusion_model.input_blocks.4.1.proj_in.weight: copying a param with shape
                         torch.Size([640, 640]) from checkpoint, the shape in current model is torch.Size([640, 640, 1,
                         1]).
18:06:32-567172 WARNING  Model not loaded


the actual model's name is "realisticStockPhoto_v10" from HuggingFace

**Reply by grendahl06** [reply]
found it on the settings tab under "Compute Settings"

for my graphics card, it seems going above 500x500 results in out of memory.  Do you know if there is a setting within the app to allow it to use my ram and CPU when I want to create a 1400x500 image?

**Comment by grendahl06** [comment]
sorry, unmarking as answered

when I restart the server, I get this error:
AttributeError: 'StableDiffusionXLPipeline' object has no attribute 'alphas_cumprod'

**Comment by paypah** [comment]
Make sure you've followed this guide when wanting to do SDXL:
https://github.com/vladmandic/automatic/wiki/SD-XL

What's your system specs? Operating System/GPU/RAM/etc.

**Reply by grendahl06** [reply]
thank you for the link.  I'm sorry to sound so incompetent, but I'm lost when it says to edit the menus.  I do not see any buttons or tabs that look like the ones on the wiki
![image](https://github.com/vladmandic/automatic/assets/28264877/285a5622-1b63-4db7-8cf0-6e47861b0768)

**Reply by paypah** [reply]
Under the "Settings" tab itself is also a "User Interface" section, that's where you'll find the options listed in the guide. The one you are on in that screenshot is a separate menu!

**Reply by grendahl06** [reply]
oh.......those....are...oh wow, that was not intuitive.  Thank you.  Attempting to follow the guide now.

**Reply by grendahl06** [reply]
it still looks like the v1 when it starts the render, but it finishes with a white image:

![image](https://github.com/vladmandic/automatic/assets/28264877/a309b257-f650-4fa9-ab7c-3465d98e06a4)

**Reply by vladmandic** [reply]
and now go back and reset the configuration. don't use refiner until you have base working and don't set pipeline to specific type, leave it at auto. i fear you got so lost in editing config that very basic setup where ONLY switching backend to diffusers was needed.

**Reply by grendahl06** [reply]
I believe I've reset the configs, including using the button in System.  Initially, the 18th of 20 iterations looks like this before turning white.

![image](https://github.com/vladmandic/automatic/assets/28264877/31901806-4a71-4b79-8c12-ae4468c59de7)

**Reply by vladmandic** [reply, marked as answer]
change sampler, its set to default which can be anything that's inside the model (or often not picked at all).
and no sdxl model works at very low resolutions. never did. use sd15 for low resolutions.

**Reply by grendahl06** [reply]
thank you.  I get the out of memory error if I use a resolution higher than this.  It's sad that with 8GB on a Radeon card, that's all I can get to work.

**Reply by vladmandic** [reply]
there are tweaks in settings you can use to reduce memory usage (default is balance between performance and resource usage). best to reach out on discord and have community help you out, i cannot go over every single setting in this thread.
and yes, there are ppl using sdxl with 8gb amd cards.

**Reply by grendahl06** [reply]
sorry, correction.  I'm not sure what is different, but I can now get 1300x600 to work with --lowvram and the associated setting changes.  1400x700 still throws a memory exception.

Thank you again

**Reply by vladmandic** [reply]
lowvram is one of those - it enables diffusers sequential cpu offload in settings which dynamically moves parts of the model from vram to ram. again, that is far too much details to go over here.
